public/app/core/utils/explore.ts

```diff
@@ -148,15 +153,17 @@ export function calculateResultsFromQueryTransactions(
     new TableModel(),
     ...queryTransactions.filter(qt => qt.resultType === 'Table' && qt.done && qt.result).map(qt => qt.result)
   );
-  const logsResult =
-    datasource && datasource.mergeStreams
+  const logsResult = {
+    ...datasource && datasource.mergeStreams
       ? datasource.mergeStreams(
           _.flatten(
             queryTransactions.filter(qt => qt.resultType === 'Logs' && qt.done && qt.result).map(qt => qt.result)
           ),
           graphInterval
         )
-      : undefined;
+      : undefined,
+    queryEmpty: queryTransactions.filter(qt => qt.resultType === 'Logs' && qt.done).every(qt => qt.result.length === 0),
```

Comment:
Is there a reason to pin this on the logs model? This can simply be calculated in render() and passed to the Logs component as a prop.

Reply:
I think initially it was, but know it isn't so I'll move it. Coming right up